lib/pipeline.js

```diff
@@ -899,12 +1004,14 @@ class PipelineModel extends BaseModel {
         this.subscribedScmUrlsWithActions = urlWithActionsList;
 
         const existingJobs = await this.pipelineJobs;
-
         const jobsProcessed = [];
         const updatedJobs = [];
         const parsedConfigJobNames = Object.keys(parsedConfig.jobs);
         const pipelineId = this.id;
 
+        // Sync pipeline stages
+        await syncStages({ pipelineJobs: existingJobs, pipelineId, stages: parsedConfig.stages || [] });
```

Comment:
Should we do a check here to see if users have `parsedStages` in their sd.yaml first before doing an `await`?

Reply:
sure